bot/exts/filtering/_ui/ui.py

```diff
@@ -531,12 +532,91 @@ async def cancel(self, interaction: Interaction, button: discord.ui.Button) -> N
         await interaction.response.edit_message(content="🚫 Operation canceled.", view=None)
 
 
+class PhishConfirmationView(discord.ui.View):
+    """Confirmation buttons for whether the alert was for a phishing attempt."""
+
+    def __init__(
+        self, mod: Member, offender: User | Member | None, phishing_content: str, target_filter_list: FilterList
+    ):
+        super().__init__()
+        self.mod = mod
+        self.offender = offender
+        self.phishing_content = phishing_content
+        self.target_filter_list = target_filter_list
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Only allow interactions from the command invoker."""
+        return interaction.user.id == self.mod.id
+
+    @discord.ui.button(label="Do it", style=discord.ButtonStyle.green)
+    async def confirm(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Auto-ban the user and add the phishing content to the appropriate filter list as an auto-ban filter."""
+        await interaction.response.edit_message(view=None)
+
+        if self.offender:
+            compban_command = bot.instance.get_command("compban")
+            if not compban_command:
+                await interaction.followup.send(':warning: Could not find the command "compban".')
+            else:
+                ctx = FakeContext(interaction.message, interaction.channel, compban_command)
```

Comment:
This makes the actor of the ban the bot, but I think it should be the moderator instead.

Reply:
Good call fa4f233